71699: Added entry on simpler plugin post_install (#636)
freenas/freenas-docs: userguide/plugins.rst

.. table:: Plugin json File Contents
   :class: longtable

   +---------------------------+--------------------------------------------------------------------+
   | Data Field                | Description                                                        |
   +===========================+====================================================================+
   | :samp:`"name":`           | Name of the plugin.                                                |
   |                           |                                                                    |
   +---------------------------+--------------------------------------------------------------------+
   | :samp:`"plugin_schema":`  | Optional. Enter *2* if simplified post install information has     |

>    |                           | been created. After specifying *2*, echo the path to the post      |

Reviewer: ```suggestion
   |                           | been supplied in :file: `post_install.sh`. After specifying *2*, echo the information to be presented to the user in :file: `/root/PLUGIN_INFO`    |
```

   |                           | install information inside the :file:`post_install.sh` file.       |
   |                           | See the :ref:`rslsync.json <rslsync-plugin-schema>` and            |
   |                           | :ref:`rslsync post_install.sh <rslsync-post-install>` examples.    |
   |                           |                                                                    |
   +---------------------------+--------------------------------------------------------------------+

>    | :samp:`"release":`        | FreeBSD release to use for the plugin jail.                        |

Reviewer: ```suggestion
   | :samp:`"release":`        | FreeBSD RELEASE to use for the plugin jail.                        |
```

   |                           |                                                                    |
   +---------------------------+--------------------------------------------------------------------+

.. _rslsync-plugin-schema:

.. code-block:: json
   :caption: rslsync.json
   :linenos:
   :emphasize-lines: 3

   {
     "name": "rslsync",
     "plugin_schema": "2",
     "release": "11.2-RELEASE",
     "artifact": "https://github.com/freenas/iocage-plugin-btsync.git",
     "pkgs": [
       "net-p2p/rslsync"
     ],
     "packagesite": "http://pkg.cdn.trueos.org/iocage/unstable",
     "fingerprints": {
	     "iocage-plugins": [
		     {
		     "function": "sha256",
		     "fingerprint": "226efd3a126fb86e71d60a37353d17f57af816d1c7ecad0623c21f0bf73eb0c7"
	     }
	     ]
     },
     "official": true
   }

.. _rslsync-post-install:

.. code-block:: sh
   :caption: post_install.sh
   :name: rslsync-post_install
   :linenos:
   :emphasize-lines: 9

   #!/bin/sh -x

   # Enable the service
   sysrc -f /etc/rc.conf rslsync_enable="YES"
   # Start the service
   service rslsync start 2>/dev/null

>    echo "Running"

Reviewer: Drop this, was debug

> .. table:: Plugin json File Contents

Reviewer 2: ```suggestion
.. table:: Plugin JSON File Contents
```

   :class: longtable

   +---------------------------+--------------------------------------------------------------------+
   | Data Field                | Description                                                        |
   +===========================+====================================================================+

>    | :samp:`"name":`           | Name of the plugin.                                                |

Reviewer 2: These should be :literal:, because they are literal text, not samples of what the user should type.

   |                           |                                                                    |
   +---------------------------+--------------------------------------------------------------------+

>    | :samp:`"plugin_schema":`  | Optional. Enter *2* if simplified post install information has     |

Reviewer 2: ```suggestion
   | :samp:`"plugin_schema":`  | Optional. Enter *2* if simplified post-install information has     |
```

   |                           | been supplied in :file:`post_install.sh`. After specifying *2*,    |
   |                           | echo the information to be presented to the user in                |
   |                           | :file:`/root/PLUGIN_INFO` inside the                               |
   |                           | :file:`post_install.sh` file.                                      |
   |                           | See the :ref:`rslsync.json <rslsync-plugin-schema>` and            |
   |                           | :ref:`rslsync post_install.sh <rslsync-post-install>` examples.    |
   |                           |                                                                    |
   +---------------------------+--------------------------------------------------------------------+
   | :samp:`"release":`        | FreeBSD RELEASE to use for the plugin jail.                        |
   |                           |                                                                    |
   +---------------------------+--------------------------------------------------------------------+

>    | :samp:`"artifact":`       | URL of the plugin artifact repo.                                   |

Reviewer 2: s/repo/repository/

   |                           |                                                                    |
   +---------------------------+--------------------------------------------------------------------+

>    | :samp:`"pkgs":`           | Port of the plugin.                                                |

Reviewer 2: Not really clear what "port" means in this context.  The description does not give any idea of the format of this data.  It could mean category/portname, but a reader might think it means a port number.

   |                           |                                                                    |
   +---------------------------+--------------------------------------------------------------------+
   | :samp:`"packagesite":`    | CDN the plugin jail uses. Default for the TrueOS CDN is            |

>    |                           | http://pkg.cdn.trueos.org/iocage .                                 |

Reviewer 2: Use :literal: around the URL, because that is the literal exact thing they are supposed to enter.

   |                           |                                                                    |
   +---------------------------+--------------------------------------------------------------------+
   | :samp:`"fingerprints":`   | :samp:`"function":`                                                |
   |                           |                                                                    |
   |                           | Default is *sha256*.                                               |
   |                           |                                                                    |

>    |                           | :samp:`"fingerprint":`                                             |

Reviewer 2: As above, when these are the literal strings the user is meant to enter, use :literal:.

   |                           |                                                                    |

>    |                           | The pkg fingerprint for the artifact repo. Default is              |

Reviewer 2: s/repo/repository/

   |                           | *226efd3a126fb86e71d60a37353d17f57af816d1c7ecad0623c21f0bf73eb0c7* |
   |                           |                                                                    |
   +---------------------------+--------------------------------------------------------------------+

>    | :samp:`"official":`       | Defines if this an official iXsystems supported plugin.            |

Reviewer 2: "Define whether this is an official iXsystems-supported plugin."
